### PR TITLE
Bump rust to 1.78.0 in actions runner

### PR DESCRIPTION
**Pull request by jpculp**
*Description of changes:*

Bumps rust to 1.78.0 and addresses the `clippy` warnings that came with it.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by bcressey** [comment]
We need to update `cargo deny` to fix this error:
```
cargo deny --all-features check --disable-fetch licenses bans sources
thread '<unnamed>' panicked at /home/runner/work/tough/tough/.cargo/registry/src/index.crates.io-6f17d22bba15001f/krates-0.14.0/src/builder.rs:791:47:
called `Option::unwrap()` on a `None` value
```

I'd prefer to keep `_extra` as-is and just add a [lints table](https://doc.rust-lang.org/cargo/reference/workspaces.html#the-lints-table) to the workspace `Cargo.toml` to allow it.

**Comment by jpculp** [comment]
- Rebased on #781 _(which updates `cargo-deny`)_.
- Added `allow(clippy::pub_underscore_fields)` to `tough/src/schema/mod.rs`.

File: tuftool/src/error.rs [hunk 1]
```diff
@@ -373,12 +351,6 @@ pub(crate) enum Error {
         backtrace: Backtrace,
     },
 
-    #[snafu(display("Failed writing target data to disk: {}", source))]
-    WriteTarget {
-        source: std::io::Error,
-        backtrace: Backtrace,
-    },
```
Comment (bcressey) [inline tuftool/src/error.rs:380]: I am surprised that so many errors end up unused.

Over in the core kit, when I was working on signpost, I noticed that a lot of the new "unused" warnings were false positives. Seems to be an issue with a crate that's built as both a library and a binary - the error variants that aren't used in the library side get flagged even though the binary needs them.

Which is to say, please double check via `git grep` or `rg` that all of these are actually not used.

**Review by bcressey — APPROVED** [review]

File: tuftool/src/error.rs (hunk 1 above)
Comment (jpculp) [reply tuftool/src/error.rs:380]: I think a couple of these might have just been copied from `tough` at some point. I wasn't able to find any references in `tuftool` itself.

**Review by cbgbt — APPROVED** [review]

**Pull request merged**